src/layers/xr-layer/xr-layer-fragment.js

```diff
@@ -75,7 +75,7 @@ void main() {
 
   for(int i = 0; i < 6; i++) {
     hsvCombo = rgb2hsv(vec3(colorValues[i]));
-    hsvCombo = vec3(hsvCombo.xy, channelArray[i]);
+    hsvCombo = vec3(hsvCombo.xy, max(0.0,channelArray[i]));
```

Comment:
This is one end of the ramp function: You're fairly confident we couldn't have a similar problem at the top of the ramp?

Reply:
For example:

```glsl
float channel0Color = (float(texture(channel0, vTexCoord).r) - sliderValues[0][0]) / sliderValues[0][1];
```

As long as  `float(texture(channel0, vTexCoord).r) - sliderValues[0][0]) < sliderValues[0][1]` we should be good on the top end. The only way this could happen, as far as I can tell, is if the data type is wrong and we get something where the sampled value is greater than the max value. It is never achieved with correct data typing (or is at worst, 1). Does that make sense?